### DISCUSSION TITLE
When creating a record, can you update a related record in the same run?

### DISCUSSION THREAD

**Discussion by aavonk** [prisma/prisma#6220] [Q&A] [answered]
While creating a new record, is it possible to update a related record in the same query using nested writes?
After looking at the docs, I see that you can _create_  a related item, but I can't seem to find anything about updates. 
Let me provide some code snippets so I can attempt to explain what I am trying to accomplish. 

Here are two of my models, Transaction and Account:

```prisma
model Transaction {
  id              String    @id @default(cuid())
  user            User      @relation(fields: [userId], references: [id])
  userId          String
  payee           String
  description     String?
  amount          Int
  category        Category? @relation(fields: [categoryId], references: [id])
  categoryId      String?
  type            String
  date            DateTime
  account         Account   @relation(fields: [accountId], references: [id])
  accountId       String
  isCashIn        Boolean   @default(false)
  isCashOut       Boolean   @default(true)
  isUncategorized Boolean   @default(true)
}
```
```prisma
model Account {
  id          String        @id @default(cuid())
  user        User          @relation(fields: [userId], references: [id])
  userId      String
  accountName String
  accountType String
  isAsset     Boolean
  isLiability Boolean
  balance     Int
  transaction Transaction[]
}
```
What I'm trying to accomplish is updating the balance of the account when a transaction is created using a nested write. Is this even possible while using ```prisma.transaction.create({...})``` or would I just have to call a separate query using the ```prisma.account.update({...})```? 

Currently, this is what my query looks like when creating a new transaction

```ts
const transaction: Transaction = await prisma.transaction.create({
      data: {
        userId: user.id,
        payee: input.payee,
        amount: input.amount,
        description: input.description,
        date: new Date(input.date),
        type: input.type,
        accountId: input.accountId,
        categoryId: input.categoryId,
        isCashIn: input.type === 'INCOME',
        isCashOut: input.type === 'EXPENSE',
        isUncategorized: !input.categoryId,
     }
      include: {
        .........
      },
      
    });
```

Is there a way to add a field to the account option that says how to update it? Something like:

```
const transaction: Transaction = await prisma.transaction.create({
      data: {
        // .... fields from above
        account: {
          update: {
            .....
          }
        }
      },
      include: {
      ...
      },
      
    });

```

**Comment by ryands17** [comment, marked as answer]
@aaronvonkreisler 👋 

You would be better of doing it the other way round. For e.g this will all be executed in a single transaction:

```ts
  await prisma.account.update({
    where: { id: 'id' },
    data: {
      balance: { decrement: 100 },
      transaction: {
        create: {
          amount: 100,
          date: new Date(),
          type: 'DEBIT',
          payee: 'Payee',
        },
      },
    },
  })
```

**Reply by aavonk** [reply]
Ahh I didn't even think of that! Thank you @ryands17

**Reply by aavonk** [reply]
@ryands17  This might diverge a bit from the original question, so let me know if I need to submit a new one, but after updating the account and creating the transaction, I'm only interested in receiving back the transaction that was just currently created. I thought that maybe I could add a "select" object to include the transaction in the response, but I get an array of all the transactions related to that account. 

My initial thought was that I could do something like this 
```ts
  await prisma.account.update({
    where: { id: 'id' },
    data: {
      balance: { decrement: 100 },
      transaction: {
        create: {
          amount: 100,
          date: new Date(),
          type: 'DEBIT',
          payee: 'Payee',
        },
      },
    },
   select: {
      transaction: {
          where: {
              id: id
         },
   },
  })
```
but obviously I am unable to access that ID as it was just created during the prisma transaction.  Is there a way to receive back just the currently created item (transaction)?

**Reply by ryands17** [reply]
Prisma returns all the related records so one. way you could do this would be fetching by the `date` of the transaction in the `where` clause as follows:

```ts
let date = new Date()
  await prisma.account.update({
    where: { id: 'id' },
    data: {
      balance: { decrement: 100 },
      transaction: {
        create: {
          amount: 100,
          date,
          type: 'DEBIT',
          payee: 'Payee',
        },
      },
    },
    include: { transaction: { where: { date } } },
})
```

**Reply by aavonk** [reply]
Perfect, thank you so much!

**Reply by bennettdams** [reply]
@ryands17 Assuming we have these two options:

1. updating/creating a nested entity "autonomously" (a transaction belongs to an account)
2. updating its parent and updating/creating the nested entity inside (like you proposed in your answer)

Is there any article/comparison for the tradeoffs between these two? I recently started with Prisma and this is something I asked myself from the beginning. Especially when working with them in the Client application, I just used option 2, because I can easier handle the new updates coming in more easily. For the example above, I just have to consume the result of the update of `account`, instead of `transaction`, as the parent already contains the updated transaction.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
